test/e2e/space_test.go

```diff
@@ -128,29 +125,106 @@ func TestSpace(t *testing.T) {
 			}
 
 			// when
-			err := hostAwait.Client.Create(context.TODO(), space)
+			err := hostAwait.Client.Create(context.TODO(), s)
 
 			// then
 			require.NoError(t, err)
-			space, err = hostAwait.WaitForSpace(space.Name, wait.UntilSpaceHasConditions(ProvisioningFailed("unknown target member cluster 'unknown'")))
+			s, err = hostAwait.WaitForSpace(s.Name, wait.UntilSpaceHasConditions(ProvisioningFailed("unknown target member cluster 'unknown'")))
 			require.NoError(t, err)
 
-			t.Run("delete space", func(t *testing.T) {
-				// now, delete the Space and expect that the NSTemplateSet will be deleted as well,
-				// along with its associated namespace
-
+			t.Run("unable to delete space", func(t *testing.T) {
 				// when
-				err = hostAwait.Client.Delete(context.TODO(), space)
+				err = hostAwait.Client.Delete(context.TODO(), s)
 
-				// then
+				// then it should fail while the member cluster is unknown (ie, unreachable)
 				require.NoError(t, err)
-				_, err = hostAwait.WaitForSpace(space.Name, wait.UntilSpaceHasConditions(TerminatingFailed("Cannot delete NSTemplateSet: unknown target member cluster: 'unknown'")))
+				s, err = hostAwait.WaitForSpace(s.Name, wait.UntilSpaceHasConditions(TerminatingFailed("cannot delete NSTemplateSet: unknown target member cluster: 'unknown'")))
 				require.NoError(t, err)
+
+				t.Run("update target cluster to unblock deletion", func(t *testing.T) {
```

Comment:
just a side note - I'm not sure if we need to verify all these error-cases also in e2e tests - they are well covered by unit tests. And here it makes the e2e test execution even longer.
To be clear, I'm fine with keeping them there - it brings another level of quality. I just wanted to point out the consequence of it, so it's up to you :wink:

Reply:
I don't think we can cover all error cases in the e2e tests, but for these particular cases, I would keep them, unless they cause more troubles than they bring benefits.

Reply:
I didn't want to cover all error cases in e2e tests ever.
Ok, if they don't cause any troubles and if we are fine with making the execution of the e2e tests longer, then I'm fine with keeping them there :+1: